compiler/rustc_codegen_llvm/src/asm.rs

```diff
@@ -642,6 +643,8 @@ fn reg_to_llvm(reg: InlineAsmRegOrRegClass, layout: Option<&TyAndLayout<'tcx>>)
             InlineAsmRegClass::Wasm(WasmInlineAsmRegClass::local) => "r",
             InlineAsmRegClass::Bpf(BpfInlineAsmRegClass::reg) => "r",
             InlineAsmRegClass::Bpf(BpfInlineAsmRegClass::wreg) => "w",
+            InlineAsmRegClass::Avr(AvrInlineAsmRegClass::reg) => "r",
+            InlineAsmRegClass::Avr(AvrInlineAsmRegClass::reg_xyz) => "r",
```

Comment:
Shouldn't this be `e`? According to the constraint list here: https://www.nongnu.org/avr-libc/user-manual/inline_asm.html

Reply:
Indeed!  Addressed.